flik/gradient/__init__.py

```diff
@@ -0,0 +1,95 @@
+# An experimental local optimization package
+# Copyright (C) 2018 Ayers Lab <ayers@mcmaster.ca>.
+#
+# This file is part of Flik.
+#
+# Flik is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# Flik is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+
+
+r"""Gradient classes."""
+
+
+from flik.objective.base import Objective
+from flik.objective.finitediff import ForwardDiffObjective
+from flik.objective.finitediff import CentralDiffObjective
+
+
+__all__ = [
+    "Gradient",
+    "ForwardDiffGradient",
+    "CentralDiffGradient",
+    ]
+
+
+class Gradient(Objective):
```

Comment:
All gradients can be objective functions for root finding.  Not all objective functions for root finding are gradients.